sw/airborne/firmwares/rotorcraft/guidance/guidance_indi.h

```diff
@@ -33,7 +33,7 @@
 #include "math/pprz_algebra_float.h"
 
 extern void guidance_indi_enter(void);
-extern void guidance_indi_run(float heading_sp);
+extern void guidance_indi_run(float *heading_sp);
```

Comment:
what is the point of making it a pointer since the value is not changed in the function ?

Reply:
indeed in guidance_indi.c the value is not changed, but when using guidance_indi_hybrid.c it is. I am now thinking that the coupling between guidance and control should even be tighter, feeding the required rates for some manoeuvre (think of a coordinated turn) to the controller (maybe it should be considered one big controller?)

Reply:
The coupling (or not) is highly dependent of the control stack. So having each layer properly separated allows a better and more flexible assembly in the end.
In your case, if you think that you need a dedicated control layer integrated to the guidance, I'm perfectly fine. If you need to only add a few options to the current stab or the ref generator, that could be done as well. I just want to avoid functions like the "set_guidance_from_nav" which is making tight coupling between nav, guidance and control from the implementation point of view, while they the design is decoupled from the control point of view.

Reply:
Thanks a lot for your insight. If I understand you correctly, you're saying that we can make modular layers for control, guidance and navigation, and big controllers that may do everything in one go. But for the modular code, it should not be interdependent. Is that a good summary? I would fully agree with that.

Reply:
Yes, that's the idea.
It will make it much easier to use the "generated" autopilot stack (see http://wiki.paparazziuav.org/wiki/Autopilot_generation).